common/buildcraft/lib/tile/TileBC_Neptune.java

```diff
@@ -375,10 +375,7 @@ protected void onSlotChange(IItemHandlerModifiable handler, int slot, @Nonnull I
      * the current chunk is saved after the last tick. */
     public void markChunkDirty() {
         if (world != null) {
-            Chunk chunk = getContainingChunk();
-            if (chunk != null) {
-                chunk.markDirty();
-            }
+            world.markChunkDirty(this.pos, this);
```

Comment:
Funnily enough this itself was an optimization - going through our cache was cheaper then calling `BlockEntity.markDirty`. I'm not sure if I tried using `World.markChunkDirty` though - since it does less work.

Reply:
Yeah, I guessed that was the point, however I figured it wasn't much of a performance concern as it wasn't showing up in profiling results at all, and the vanilla version of `World.markChunkDirty` isn't too much different than the existing one (hashtable lookup through `ChunkProviderServer` instead of accessing the cached chunk instance, but presumably negligible impact if I already couldn't measure it before). 

However in Cubic Chunks worlds `World.markChunkDirty` actually marks the *cube* as dirty, which is important because otherwise the cube might not get saved. (I actually saw this happen in practice - a BuildCraft quarry in a Cubic Chunks world piping items into a crate from another mod which was also not using `World.markChunkDirty` resulted in the mined blocks not being saved, even though the blocks broken by the quarry were!)

If you're not certain about this change I can totally move it into a separate PR. I just threw it in here since I figured it wouldn't make too much difference, but wouldn't mind profiling it in more depth if you'd like.

Reply:
Back in my day, what we would do is put certain optimizations behind compatibility flags (f.e. we had an important optimization in pipes that broke on Cauldron-based servers). Maybe this would be the way to go here?